File: 1. Basics/A_Beautiful_Matrix.rs
use std::io;

fn main() {
    let mut matrix = [[0; 5]; 5];

    for i in 0..5 {
        let mut input = String::new();
        io::stdin().read_line(&mut input).unwrap();
        let nums: Vec<i32> = input.trim().split_whitespace().map(|x| x.parse().unwrap()).collect();
        
        for j in 0..5 {
            matrix[i][j] = nums[j];
        }
    }

    for i in 0..5 {
        for j in 0..5 {
            if matrix[i][j] == 1 {
                let moves = (i as i32 - 2).abs() + (j as i32 - 2).abs();
                println!("{moves}");
            }
        }
    }
}

File: Basics/A_Beautiful_Matrix.py
matrix = []

for i in range(5):
    row = list(map(int, input().split()))
    matrix.append(row)

pos = []
for i in range(len(matrix)):
    for j in range(len(matrix[0])):
        if matrix[i][j] == 1:
            pos = [i+1,j+1]

print(abs(3-pos[0]) + abs(3-pos[1]))